src/generic/AhoCorasick.jl

```diff
@@ -0,0 +1,214 @@
+#
+#   FreeAssAhoCorasick.jl : implement bulk divide check for leading terms of free associative Algebra elements
+#   for use e.g. in Groebner Basis computation
+#
+###############################################################################
+
+export aho_corasick_automaton
+
+export AhoCorasickAutomaton
+
+export AhoCorasickMatch
+
+export insert_keyword!
+
+export search
+
+#export Word
+#
+import DataStructures: Queue, enqueue!, dequeue!
```

Comment:
So we add a dependency on `DataStructures` just for a `Queue` type? I am not sure this will fly. Is this a crucial bottleneck for you? Otherwise I'd suggest a simple custom implementation here? Like
```
struct Queue{T}
  data::Vector{T}
end

function enqueue!(q::Queue{T}, val::T) where T
  push!(q.data, val)
end
function dequeue!(q::Queue)
  return popfirst!(q.data)
end
isempty(q::Queue) = isempty(q.data)

```
well and perhaps some extra bells&whistles depending on your need. Of course this is suboptimal, but it may not matter for your use case? And if it matters, we can easily insert a better implementation.

Reply:
Also, do you really need a queue? Or would a stack also work? Then you can just use a plain `Vector` with `push!` and `pop!`

Reply:
I'm mainly using `DataStructures` for a `PriorityQueue` in FreeAssAlgebraGroebner.jl. At least on examples that terminated, this perceivably increased the performance.